HIP/hip-865.md

```diff
@@ -0,0 +1,104 @@
+---
+hip: 865
+title: Add EVM Support for transient storage and memory copy Cancun opcodes
+author: Danno Ferrin (@shemnon)
+working-group: Nana Essilfie-Conduah <@nana-ec>
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Draft
+created: 2024-01-24
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/872
+---
+
+## Abstract
+
+Update the Hedera network to add support for new Opcodes found in the Cancun fork of
+Ethereum Mainnet, namely TSTORE, TLOAD, and MCOPY
+
+## Motivation
+
+Hedera's goal of EVM Equivalence also includes the requirement to stay
+up-to-date with the current state of Ethereum Mainnet. This HIP addresses new
+opcodes added to the EVM that do not address any features relating to Blobs, or
+the Consensus Layer, or changes to existing opcodes.
+
+## Rationale
+
+The opcodes discussed in this HIP only exist within the EVM and do not interact
+with any novel features of ethereum mainnet (such as blobs and CL beacon root).
+It is expected that future versions of solidity, Vyper, and other languages
+targeting the EVM will support both implicit and explicit use of these opcodes.
+Because of that we need to support them as specified.
+
+## User stories
+
+* As a smart contract author, I want to be able to use transient storage
+  features of solidity in my hedera smart contracts.
+* As a smart contract author, I want to be able to use future versions of
+  solidity that may use memory copying implicitly.
+* As a smart contract author, I want to be able to explicitly use the memory
+  copy features solidity may expose in future releases.
+
+## Specification
+
+### EVM Implementation
+
+Two EIPs define the operational semantics of the added opcodes. For the
+transient storage opcodes `TSTORE` and `TLOAD` they are defined
+in [EIP-1153](https://eips.ethereum.org/EIPS/eip-1153). For the `MCOPY`
+operation they are defined
+in [EIP-5656](https://eips.ethereum.org/EIPS/eip-5656).
+
+The specified opcodes are to be implemented identically to Ethereum Mainnet and
+as specified in their respective EIPs. This includes opcode numbers, gas
+schedules, stack semantics, and new facilities such as transient storage added
+to the execution frame.
+
+## Backwards Compatibility
+
+The core EVM library shipping with Hedera as of version 0.46 already contains
+the needed EVM support. The activation will add a new Hedera EVM version that
+will activate all the Cancun support in one release.
+
+## Security Implications
+
+Because the operations are being brought in with identical semantics there are
+no security risks above those already present from existing Ethereum Equivalence
+changes.
+
+## How to Teach This
+
+Any smart contract tutorials will want to examine the possibility of adding
+sample contracts showcasing the use of transient storage and easy memory
```

Comment:
Should we mention that `MCOPY` will have more impact on compilers and low-level assembly devs rather than Solidity devs?

Reply:
There are explicit ways to trigger mcopy, such as assembly blocks.  Gas golfers will want to see it.